tests/v2/provisioining/provisioining_harvester_node_driver_test.go

```diff
@@ -0,0 +1,222 @@
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+}
+
+func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioning_RKE2HarvesterCluster() {
+	subSession := r.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := r.client.WithSession(subSession)
+	require.NoError(r.T(), err)
+
+	cloudCredential, err := harvester.CreateHarvesterCloudCredentials(client)
+	require.NoError(r.T(), err)
+
+	nodeRoles0 := []map[string]bool{
+		{
+			"controlplane": true,
+			"etcd":         true,
+			"worker":       true,
+		},
+	}
+
+	nodeRoles1 := []map[string]bool{
+		{
+			"controlplane": true,
+			"etcd":         false,
+			"worker":       false,
+		},
+		{
+			"controlplane": false,
+			"etcd":         true,
+			"worker":       false,
+		},
+		{
+			"controlplane": false,
+			"etcd":         false,
+			"worker":       true,
+		},
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []map[string]bool
+		client    *rancher.Client
+	}{
+		{"1 Node all roles Admin User", nodeRoles0, r.client},
+		{"1 Node all roles Standard User", nodeRoles0, r.standardUserClient},
+		{"3 nodes - 1 role per node Admin User", nodeRoles1, r.client},
+		{"3 nodes - 1 role per node Standard User", nodeRoles1, r.standardUserClient},
+	}
+
+	for _, tt := range tests {
+		r.Run(tt.name, func() {
+			testSession := session.NewSession(r.T())
+			defer testSession.Cleanup()
+
+			testSessionClient, err := tt.client.WithSession(testSession)
+			require.NoError(r.T(), err)
+
+			clusterName := GenerateRandomName(baseHarvesterClusterName)
+			generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+			machinePoolConfig := machinepools.NewHarvesterMachineConfig(generatedPoolName, "default", "default/ctw-network-1", "default/image-rpj98")
+
+			machineConfigResp, err := machinepools.CreateMachineConfig(machinepools.HarvesterResourceConfig, machinePoolConfig, testSessionClient)
+			require.NoError(r.T(), err)
+
+			machinePools := machinepools.RKEMachinePoolSetup(tt.nodeRoles, machineConfigResp)
+
+			cluster := clusters.NewRKE2ClusterConfig(clusterName, "fleet-default", "calico", cloudCredential.ID, "v1.21.7+rke2r2", machinePools)
+
+			clusterResp, err := clusters.CreateRKE2Cluster(testSessionClient, cluster)
+			require.NoError(r.T(), err)
+
+			result, err := testSessionClient.Provisioning.Clusters("fleet-default").Watch(context.TODO(), metav1.ListOptions{
+				FieldSelector:  "metadata.name=" + clusterName,
+				TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+			})
+			require.NoError(r.T(), err)
+
+			checkFunc := clusters.IsProvisioningClusterReady
+
+			err = wait.WatchWait(result, checkFunc)
+			assert.NoError(r.T(), err)
+			assert.Equal(r.T(), clusterName, clusterResp.Name)
+
+		})
+	}
+}
+
+// func (r *RKE2NodeDriverProvisioningTestSuite) TestProvisioning_RKE2HarvesterClusterDynamicInput() {
+// 	nodesAndRoles := NodesAndRolesInput()
+// 	if len(nodesAndRoles) == 0 {
+// 		r.T().Skip()
+// 	}
+
```

Comment:
Did you test out with the node inputs? Once we get my PR in, for sure want to have this working. Also I think these tests can go in provisioning_node_driver_test.go, once we get my PR merged in.